junit-resource-poller/src/main/java/com/palantir/junit/HttpPollingResource.java

```diff
@@ -44,28 +45,91 @@
-    public static HttpPollingResource of(
-            Optional<SSLSocketFactory> socketFactory, String pollUrl, int numAttempts) {
-        return new HttpPollingResource(socketFactory, Collections.singletonList(pollUrl), numAttempts, 100,
+    public static HttpPollingResource create(
+            Optional<SslParameters> sslParameters, String pollUrl, int numAttempts) {
+        return new HttpPollingResource(sslParameters, Collections.singletonList(pollUrl), numAttempts, 100,
+                CONNECTION_TIMEOUT_MILLIS, READ_TIMEOUT_MILLIS);
+    }
+
+    /**
+     * Like {@link HttpPollingResource#create(Optional, String, int)}, but waits for all of the given URLs.
+     */
+    public static HttpPollingResource create(
+            Optional<SslParameters> sslParameters, Collection<String> pollUrls, int numAttempts) {
+        return new HttpPollingResource(sslParameters, pollUrls, numAttempts, 100,
                 CONNECTION_TIMEOUT_MILLIS, READ_TIMEOUT_MILLIS);
     }
 
+    /**
+     * Waits at most {@code timeout} until a GET request for {@code pollUrl} succeeds, polls every 100 milliseconds.
+     *
+     * @deprecated Use com.palantir.junit.HttpPollingResource#create(java.util.Optional, java.util.Collection, int)
+     */
+    @Deprecated
+    public static HttpPollingResource of(
+            Optional<SSLSocketFactory> sslSocketFactory, String pollUrl, int numAttempts) {
+        return new HttpPollingResource(
+                sslSocketFactory.map(SslParameters::of),
+                Collections.singletonList(pollUrl),
+                numAttempts,
+                100,
+                CONNECTION_TIMEOUT_MILLIS,
+                READ_TIMEOUT_MILLIS);
+    }
+
     /**
      * Like {@link HttpPollingResource#of(Optional, String, int)}, but waits for all of the given URLs.
+     *
+     * @deprecated Use com.palantir.junit.HttpPollingResource#create(java.util.Optional, java.util.Collection, int)
      */
+    @Deprecated
     public static HttpPollingResource of(
-            Optional<SSLSocketFactory> socketFactory, Collection<String> pollUrls, int numAttempts) {
-        return new HttpPollingResource(socketFactory, pollUrls, numAttempts, 100,
-                CONNECTION_TIMEOUT_MILLIS, READ_TIMEOUT_MILLIS);
+            Optional<SSLSocketFactory> sslSocketFactory, Collection<String> pollUrls, int numAttempts) {
+        return new HttpPollingResource(
+                sslSocketFactory.map(SslParameters::of),
+                pollUrls,
+                numAttempts,
+                100,
+                CONNECTION_TIMEOUT_MILLIS,
+                READ_TIMEOUT_MILLIS);
     }
 
-    public HttpPollingResource(Optional<SSLSocketFactory> socketFactory, Collection<String> pollRequests,
-            int numAttempts, long intervalMillis, int connectionTimeoutMillis, int readTimeoutMillis) {
+    public HttpPollingResource(
+            Optional<SslParameters> sslParameters,
```

Comment:
This `SSLSocketFactory -> SslParameters` change broke Java API and means automatic PRs don't go in without manual attention :/

do you mind if I bring back the backcompat thing?

Reply:
My bad, I didn't realize that the constructor was also public - I thought only the static factory was.

Reply:
The problem is the two constructors (`Optional<SslParameters> and `Optional<SSLSocketFactory>`) can't be overloaded, because they have the same erasure. To reintroduce the old constructor, we'd have to break people who moved to the new constructor.

Reply:
Yep I think bringing back the old consumers is the right approach here. There are tons of usages of 0.4.0 but since 0.5.0 has only been released for a couple of days (and didn't auto upgrade) I think it's ok to call this a 'fix'.

Reply:
https://github.com/palantir/junit-resource-poller/pull/13